packages/tailwindcss/src/css-functions.test.ts

```diff
@@ -152,14 +152,212 @@ describe('--theme(…)', () => {
+      `),
+    ).toMatchInlineSnapshot(`
+      ":root, :host {
+        --color-red-500: red;
+      }
+
+      .red {
+        color: color-mix(in oklab, var(--color-red-500) 50%, transparent);
+      }"
+    `)
+  })
+
+  test('--theme(--color-red-500/50 inline)', async () => {
+    expect(
+      await compileCss(css`
+        @theme {
+          --color-red-500: #f00;
+        }
+        .red {
+          color: --theme(--color-red-500/50 inline);
+        }
+      `),
+    ).toMatchInlineSnapshot(`
+      ".red {
+        color: oklab(62.7955% .224863 .125846);
+      }"
+    `)
+  })
+
+  test('--theme(…) injects the fallback when the value it refers is set to a `--theme(…)` function containing `initial`', async () => {
+    expect(
+      await compileCss(css`
+        @theme prefix(tw) {
+          --default-font-family: --theme(--font-family, initial);
+        }
+        .red {
+          font-family: --theme(--default-font-family, Potato Sans, sans-serif);
+        }
+      `),
+    ).toMatchInlineSnapshot(`
+      ".red {
+        font-family: var(--tw-default-font-family, Potato Sans, sans-serif);
+      }"
+    `)
+  })
+
+  test('--theme(…) injects the fallback when the value it refers is set to a `--theme(…)` function containing `initial` in @reference mode', async () => {
+    expect(
+      await compileCss(css`
+        @theme reference prefix(tw) {
+          --default-font-family: --theme(--font-family, initial);
+        }
+        .red {
+          font-family: --theme(--default-font-family, Potato Sans, sans-serif);
+        }
+      `),
+    ).toMatchInlineSnapshot(`
+      ".red {
+        font-family: var(--tw-default-font-family, Potato Sans, sans-serif);
+      }"
+    `)
+  })
+
+  test('--theme(…) returns the fallback when the value it refers is set to a `--theme(…)` function that returns `initial` as inline value', async () => {
+    expect(
+      await compileCss(css`
+        @theme prefix(tw) {
+          --default-font-family: --theme(--font-family inline, initial);
+        }
+        .red {
+          font-family: --theme(--default-font-family inline, Potato Sans, sans-serif);
+        }
+      `),
+    ).toMatchInlineSnapshot(`
+      ".red {
+        font-family: Potato Sans, sans-serif;
+      }"
+    `)
+  })
+
+  test('--theme(…) returns the fallback when the value it refers is set to a `--theme(…)` function that returns `initial` as inline value in @reference mode', async () => {
+    expect(
+      await compileCss(css`
+        @theme reference prefix(tw) {
+          --default-font-family: --theme(--font-family, initial);
+        }
+        .red {
+          font-family: --theme(--default-font-family, Potato Sans, sans-serif);
+        }
+      `),
+    ).toMatchInlineSnapshot(`
+      ".red {
+        font-family: var(--tw-default-font-family, Potato Sans, sans-serif);
+      }"
+    `)
+  })
+
+  test('--theme(…) does not inject the fallback if the fallback is `initial`', async () => {
```

Comment:
This test name feels wrong. The different thing about this test is that the theme key exists and therefore the fallback isn't used.

Reply:
good catch!